[cmd/opampsupervisor] Test TestValidate/Valid_filled_out_config failure

### Component(s)

cmd/opampsupervisor

### Describe the issue you're reporting

[Failing CI/CD link](https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/9418887679/job/25947369585?pr=33420)

Note: This failure occurred on Windows, not sure if it's a Windows-specific issue or not.

Failure output:
```
✖  supervisor/config (68ms)

=== Failed
=== FAIL: supervisor/config TestValidate/Valid_filled_out_config (0.00s)
    config_test.go:281: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor/config/config_test.go:281
        	Error:      	Received unexpected error:
        	            	agent::executable does not have executable bit set
        	Test:       	TestValidate/Valid_filled_out_config

make[2]: *** [../../Makefile.Common:126: test] Error 1
=== FAIL: supervisor/config TestValidate (0.06s)

=== FAIL: supervisor/config TestValidate/Valid_filled_out_config (re-run 1) (0.00s)
    config_test.go:281: 
        	Error Trace:	D:/a/opentelemetry-collector-contrib/opentelemetry-collector-contrib/cmd/opampsupervisor/supervisor/config/config_test.go:281
        	Error:      	Received unexpected error:
        	            	agent::executable does not have executable bit set
make[1]: *** [Makefile:179: cmd/opampsupervisor] Error 2
        	Test:       	TestValidate/Valid_filled_out_config
make[1]: *** Waiting for unfinished jobs....

=== FAIL: supervisor/config TestValidate (re-run 1) (0.00s)
```

--- a/cmd/opampsupervisor/supervisor/config/config.go
+++ b/cmd/opampsupervisor/supervisor/config/config.go
@@ -132,13 +132,9 @@ func (a Agent) Validate() error {
 		return errors.New("agent::executable must be specified")
 	}
 
-	f, err := os.Stat(a.Executable)
+	_, err := os.Stat(a.Executable)
 	if err != nil {
 		return fmt.Errorf("could not stat agent::executable path: %w", err)
-	}
-
-	if f.Mode().Perm()&0111 == 0 {
-		return fmt.Errorf("agent::executable does not have executable bit set")
 	}
 
 	return nil

--- a/cmd/opampsupervisor/supervisor/config/config_test.go
+++ b/cmd/opampsupervisor/supervisor/config/config_test.go
@@ -34,7 +34,7 @@ func TestValidate(t *testing.T) {
 					},
 				},
 				Agent: Agent{
-					Executable:              "${executable_path}",
+					Executable:              "${file_path}",
 					OrphanDetectionInterval: 5 * time.Second,
 				},
 				Capabilities: Capabilities{
@@ -57,7 +57,7 @@ func TestValidate(t *testing.T) {
 					},
 				},
 				Agent: Agent{
-					Executable:              "${executable_path}",
+					Executable:              "${file_path}",
 					OrphanDetectionInterval: 5 * time.Second,
 				},
 				Capabilities: Capabilities{
@@ -82,7 +82,7 @@ func TestValidate(t *testing.T) {
 					},
 				},
 				Agent: Agent{
-					Executable:              "${executable_path}",
+					Executable:              "${file_path}",
 					OrphanDetectionInterval: 5 * time.Second,
 				},
 				Capabilities: Capabilities{
@@ -107,7 +107,7 @@ func TestValidate(t *testing.T) {
 					},
 				},
 				Agent: Agent{
-					Executable:              "${executable_path}",
+					Executable:              "${file_path}",
 					OrphanDetectionInterval: 5 * time.Second,
 				},
 				Capabilities: Capabilities{
@@ -136,7 +136,7 @@ func TestValidate(t *testing.T) {
 					},
 				},
 				Agent: Agent{
-					Executable:              "${executable_path}",
+					Executable:              "${file_path}",
 					OrphanDetectionInterval: 5 * time.Second,
 				},
 				Capabilities: Capabilities{
@@ -199,31 +199,6 @@ func TestValidate(t *testing.T) {
 			expectedError: "could not stat agent::executable path:",
 		},
 		{
-			name: "agent executable has no exec bits set",
-			config: Supervisor{
-				Server: OpAMPServer{
-					Endpoint: "wss://localhost:9090/opamp",
-					Headers: http.Header{
-						"Header1": []string{"HeaderValue"},
-					},
-					TLSSetting: configtls.ClientConfig{
-						Insecure: true,
-					},
-				},
-				Agent: Agent{
-					Executable:              "${non_executable_path}",
-					OrphanDetectionInterval: 5 * time.Second,
-				},
-				Capabilities: Capabilities{
-					AcceptsRemoteConfig: true,
-				},
-				Storage: Storage{
-					Directory: "/etc/opamp-supervisor/storage",
-				},
-			},
-			expectedError: "agent::executable does not have executable bit set",
-		},
-		{
 			name: "Invalid orphan detection interval",
 			config: Supervisor{
 				Server: OpAMPServer{
@@ -236,7 +211,7 @@ func TestValidate(t *testing.T) {
 					},
 				},
 				Agent: Agent{
-					Executable:              "${executable_path}",
+					Executable:              "${file_path}",
 					OrphanDetectionInterval: -1,
 				},
 				Capabilities: Capabilities{
@@ -253,25 +228,17 @@ func TestValidate(t *testing.T) {
 	// create some fake files for validating agent config
 	tmpDir := t.TempDir()
 
-	executablePath := filepath.Join(tmpDir, "agent.exe")
-	//#nosec G306 -- need to write executable file for test
-	require.NoError(t, os.WriteFile(executablePath, []byte{}, 0100))
-
-	nonExecutablePath := filepath.Join(tmpDir, "file")
-	require.NoError(t, os.WriteFile(nonExecutablePath, []byte{}, 0600))
+	filePath := filepath.Join(tmpDir, "file")
+	require.NoError(t, os.WriteFile(filePath, []byte{}, 0600))
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
 			// Fill in path to agent executable
 			tc.config.Agent.Executable = os.Expand(tc.config.Agent.Executable,
 				func(s string) string {
-					switch s {
-					case "executable_path":
-						return executablePath
-					case "non_executable_path":
-						return nonExecutablePath
+					if s == "file_path" {
+						return filePath
 					}
-
 					return ""
 				})
 

